text-serializer-ansi/src/main/java/net/kyori/adventure/text/serializer/ansi/ANSIComponentSerializerImpl.java

```diff
@@ -0,0 +1,276 @@
+ * {@inheritDoc}
+ */
+final class ANSIComponentSerializerImpl implements ANSIComponentSerializer {
+  private final boolean colorDownSample;
+  static ANSIComponentSerializer TRUE_COLOR = new ANSIComponentSerializerImpl(false);
+
+  private ANSIComponentSerializerImpl(final boolean colorDownSample) {
+    this.colorDownSample = colorDownSample;
+  }
+
+  private static final TextDecoration[] DECORATIONS = TextDecoration.values();
+  private static final String RGB_FORMAT = ESC_CHAR + "[38;2;%d;%d;%dm";
+  private static final Pattern STRIP_ESC_CODES = Pattern.compile(ESC_CHAR + "\\[[;\\d]*m");
+
+  private Ansi toANSI(final TextFormat format){
+    if(format instanceof NamedTextColor || format instanceof TextDecoration){
+      return this.ansiFormat(Formats.byFormat(format));
+    }
+    if(format instanceof TextColor){
+      if(this.colorDownSample){
+        return this.ansiFormat(Formats.byFormat(NamedTextColor.nearestTo((TextColor) format)));
+      }
+      final int red = ((TextColor) format).red();
+      final int blue = ((TextColor) format).blue();
+      final int green = ((TextColor) format).green();
+      return Ansi.ansi().format(RGB_FORMAT, red, green, blue);
+    }else{
+      return null;
+    }
+  }
+
+  private @Nullable Ansi ansiFormat(final @Nullable Formats formats){
+    if(formats == null){
+      return null;
+    }else{
+      return formats.ansi();
+    }
+  }
+
+  /**
+   * Deserializes a ansi string into an output of type {@link Component}. This
+   * <b>does not</b> covert ansi colours to styles.
+   *
+   * @param input the input
+   * @return {@link TextComponent}
+   */
+  @Override
+  public @NonNull Component deserialize(final @NonNull String input) {
+    final Matcher matcher = STRIP_ESC_CODES.matcher(input);
+    String out = input;
+    while(matcher.find()){
+      out = out.replace(matcher.group(), "");
+    }
+    return Component.text(out);
+  }
+
+  /**
+   * Serializes a component into an output of type {@link ANSIComponentSerializer}.
+   *
+   * @param component the component
+   * @return the output
+   */
+  @Override
+  public @NonNull String serialize(final @NonNull Component component){
+    final Porridge state = new Porridge();
+    state.append(component);
+    return state.toString();
+  }
+
+  @Override
+  public @NonNull Builder toBuilder(){
+    return new BuilderImpl(this);
+  }
+
+  static final class BuilderImpl implements Builder {
+    private boolean colorDownSample = true;
+
+    BuilderImpl(){
+
+    }
+
+    BuilderImpl(final @NonNull ANSIComponentSerializerImpl serializer){
+      this.colorDownSample = serializer.colorDownSample;
+    }
+
+    @Override
+    public @NonNull Builder downSample(final boolean downSample) {
+      this.colorDownSample = downSample;
+      return this;
+    }
+
+    @Override
+    public @NonNull ANSIComponentSerializer build() {
+      return new ANSIComponentSerializerImpl(this.colorDownSample);
+    }
+  }
+
+  private final class Porridge {
```

Comment:
Cool names are cool, but it's harder to see what this does or why it's needed.

Reply:
It was for exactly the same reason as the old class in the Legacy serializer was called Cereal....it was messy - so I named mine in a similar fashion - to be consistent...my argument was mine was a hot mess.  That being said its now named StyleSetter again for consistency.